src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -302,6 +302,12 @@ public StorageClientFactory StorageClientFactory
             }
         }
 
+        // Test hook to set storage account. 
+        internal void SetAccountProvider(IStorageAccountProvider provider)
```

Comment:
I don't think this method should be added. There is already a mechanism to do this, by setting the ContextFactory property. That's what other tests do. This doesn't feel like a pattern we want to promote - adding a new setter for every new thing a test wants to mock.